Tensile/Source/lib/include/Tensile/ContractionSolution.hpp

```diff
@@ -286,11 +287,11 @@ namespace Tensile
             double max       = 1000.0;
         };
 
-        int         index = 0;
-        std::string kernelName;
-        std::string codeObjectFilename;
-        bool        debugKernel   = false;
-        bool        kernelArgsLog = false;
+        int                          index = 0;
+        std::string                  kernelName;
+        ThreadSafeValue<std::string> codeObjectFilename;
```

Comment:
What's the difference between this and `std::atomic<std::string>`?

Reply:
Oh. `ThreadSafeValue<std::string>` is copyable while `std::atomic<std::string>` is not. I'd suggest explicitly contrasting the two in the documentation.

Reply:
You need it to be a trivially copyable type to use atomic. The alternative would be using an atomic pointer and adding copy/assignment/destructors/etc to delete it properly